src/components/cardcollection/CardCollectionSecondary.java

```diff
@@ -0,0 +1,47 @@
+package components.cardcollection;
+
+/**
+ * Layered implementation of secondary methods for {@code CardCollection}.
+ */
+public class CardCollectionSecondary implements CardCollection {
+
+    @Override
+    public final int valueBWDates(String date1, String date2) {
+        int value1 = this.valAtDate(date1), value2 = this.valAtDate(date2);
+
+        return (value2 - value1);
+    }
+
+    @Override
+    public final void changeDate(String originalDate, String date2) {
+        int valAtOgDate = this.removeCardDate(originalDate);
+        this.addCardValue(date2, valAtOgDate);
+    }
+
+    @Override
+    public final boolean valBigger(String date1, String date2) {
+        int val1 = this.valAtDate(date1), val2 = this.valAtDate(date2);
+
+        return (val1 > val2);
+    }
+
+    @Override
+    public final boolean equals(Object an) {
+        boolean same = true;
```

Comment:
Looks good. See the common methods slides for a better way to implement this method.